execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -1171,21 +1171,29 @@ where
     }
 
     /// Advances eras by num_eras
-    pub fn advance_eras_by(&mut self, num_eras: u64) {
-        for _ in 0..=num_eras {
-            let step_request = StepRequestBuilder::new()
+    pub fn advance_eras_by(
+        &mut self,
+        num_eras: u64,
+        configure: Option<fn(StepRequestBuilder) -> StepRequestBuilder>,
+    ) {
```

Comment:
Personally, I don't like methods that take closure too much. I think extending the `advance_eras_by` (and `advance_eras_by_default_auction_delay`) with `rewards` (which could be of type `Iterator<Item=T>, T: Into<RewardItem>`) and `slashes` is still fine. @Fraser999 , wdyt?

Reply:
I agree about disliking the optional closure.

I don't think I'd try and handle that via these two helper methods, as it's not obvious which step to add those rewards and slashes to.  I'd either just not try to handle that via these helpers (and leave that code as it is in `dev`), or since it seems like these args are only used when advancing by a single era, maybe provide a new helper which only advances by a single era, but takes the two collections as args.